src/Datadog.Trace/AsyncLocalScopeManager.cs

```diff
@@ -8,26 +10,46 @@ internal class AsyncLocalScopeManager : IScopeManager
 
         private readonly AsyncLocalCompat<Scope> _activeScope = new AsyncLocalCompat<Scope>();
 
+        public event EventHandler<ScopeEventArgs> ScopeOpened;
+
+        public event EventHandler<ScopeEventArgs> ScopeActivated;
+
+        public event EventHandler<ScopeEventArgs> ScopeDeactivated;
+
+        public event EventHandler<ScopeEventArgs> ScopeClosed;
```

Comment:
So is the idea for the language:
Opened - A scope is instantiated (i.e., `new Scope()`)
Activated - a scope becomes the "active" scope (e.g., the scope was "opened" or a child "closes")
Deactivated - a scope loses it's place as "active" scope (e.g., a child "opens" or a scope "closes")
Closed - A scope is closed (i.e., the active `Scope` is disposed)

Reply:
Yes that's exactly it